packages/convex-helpers/server/zod.ts

```diff
@@ -570,11 +571,8 @@ type ConvexValidatorFromZod<Z extends z.ZodTypeAny> =
                                                   ConvexValidatorFromZod<Inner>["fieldPaths"]
                                                 >
                                               : never
-                                          : Z extends z.ZodBranded<
-                                                infer Inner,
-                                                any
-                                              >
-                                            ? ConvexValidatorFromZod<Inner>
+                                          : Z extends z.ZodBranded<z.ZodString, infer Brand>
+                                            ? VString<string & { _: Brand }>
```

Comment:
I think you need to use the `BRAND` from `zod` (https://github.com/colinhacks/zod/blob/1f4f0dacf313a2dba45563d78171e6f016096925/src/types.ts#L4904-L4905) here:

```suggestion
                                          : Z extends z.ZodBranded<z.ZodString, infer Brand>
                                            ? VString<string & BRAND<Brand>>
```

Maybe it would be better to also support branding types other than just strings?